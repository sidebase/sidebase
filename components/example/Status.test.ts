import { afterEach, describe, expect, test, vi } from 'vitest'
import Status from './Status.vue'
import { render } from '~/tests/testingLibraryVue'
import type { ResponseHealthcheck } from '~/server/schemas/healthz'

describe('Status', () => {
  afterEach(() => {
    vi.restoreAllMocks()
  })

  test('mounts and is still the same for fail variant', () => {
    const { html } = render(Status, { props: { statusCheck: null } })
    expect(html()).toMatchSnapshot()
  })

  test('mounts and is still the same for success variant', () => {
    const timeToUse = new Date(2000, 1, 1, 14)
    const timeToUseAsFixedLocale = timeToUse.toLocaleString('en-US')
    const fakeStatusCheckSuccess: ResponseHealthcheck = {
      status: 'healthy',
      startupTime: timeToUse,
      time: timeToUse,
      nuxtAppVersion: 'v0.0.1',
    }

    // Render with mocked locale, so that formatting is also the same in CI / on different continents / ...
    vi.spyOn(global.Date.prototype, 'toLocaleString').mockImplementation(() => timeToUseAsFixedLocale)
    const { html } = render(Status, { props: { statusCheck: fakeStatusCheckSuccess } })

    expect(html()).toMatchSnapshot()
  })

  test('displays all status values', async () => {
    const statusCheck: ResponseHealthcheck = {
      status: 'healthy',
      startupTime: new Date(2000, 1, 1, 13),
      time: new Date(2000, 1, 1, 14),
      nuxtAppVersion: 'v0.0.1',
    }

    const { getByText } = render(Status, { props: { statusCheck } })

    // Ensure all expected text is there:
    getByText('Live backend service status is:')
    getByText('Status')
    getByText('healthy')
    getByText(`${statusCheck.time.toLocaleString()}`)
    getByText(`${statusCheck.startupTime.toLocaleString()}`)
    getByText(`${statusCheck.nuxtAppVersion}`)
  })
})
