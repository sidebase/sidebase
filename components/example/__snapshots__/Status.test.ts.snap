// Vitest Snapshot v1

exports[`Status > mounts and is still the same for fail variant 1`] = `
"<div>
  <p class=\\"font-bold\\"> An error occurred. Health status could not be requested. </p>
</div>"
`;

exports[`Status > mounts and is still the same for success variant 1`] = `
"<div>
  <div>
    <div> Live backend service status is: </div>
    <table class=\\"w-full table-fixed\\">
      <tbody>
        <tr>
          <td> Status </td>
          <td>healthy</td>
        </tr>
        <tr>
          <td>Server time</td>
          <td>2/1/2000, 2:00:00 PM</td>
        </tr>
        <tr>
          <td>Startup Time</td>
          <td>2/1/2000, 2:00:00 PM</td>
        </tr>
        <tr>
          <td>App version</td>
          <td>v0.0.1</td>
        </tr>
      </tbody>
    </table>
    <!--v-if-->
  </div>
</div>"
`;
